systemtest/src/main/java/io/strimzi/systemtest/utils/ClientUtils.java

```diff
@@ -24,18 +23,21 @@
     // ensuring that object can not be created outside of class
     private ClientUtils() {}
 
-    public static void waitUntilClientReceivedMessagesTls(KafkaClientOperations kafkaClient, int exceptedMessages) {
-        TestUtils.waitFor("Kafka " + kafkaClient.toString() + " client received messages", Constants.GLOBAL_CLIENTS_POLL, Constants.GLOBAL_TIMEOUT,
-            () -> {
-                int receivedMessages = 0;
-                try {
-                    receivedMessages = kafkaClient.receiveMessagesTls(Constants.GLOBAL_CLIENTS_TIMEOUT);
-                    return receivedMessages == exceptedMessages;
-                } catch (Exception e) {
-                    LOGGER.warn("Client not received excepted messages {}, instead received only {}!", exceptedMessages, receivedMessages);
-                    return false;
-                }
-            });
+    public static void waitUntilClientReceivedMessagesTls(KafkaClientOperations kafkaClient, int exceptedMessages) throws Throwable {
+        int receivedMessages;
+        int tries = 3;
+
+        for (; tries > 0; tries--) {
```

Comment:
why do you reverse the counting? Shouldn't be better iterate from 1 to 3 inclusive and use that value in the logging?